### DISCUSSION TITLE
Cross-compiling Oxigraph for ARM devices

### DISCUSSION THREAD

**Discussion by cravishe** [oxigraph/oxigraph#405] [Q&A]
I want to run oxigraph (rust) in RPI device. Can oxigraph be built (using rustc or cargo build) for ARM architecture? Has it been tried before?

**Comment by Tpt** [comment]
Hi @chiru091096! Yes, Oxigraph works on 64-bits ARM. I had no problem using it on a RPI 4. There are pre-built wheels for the Python bindings on Pypi and I just added last week a task to the CI/CD to build server binaries. Here is [the latest nightly build](https://github.com/oxigraph/oxigraph/suites/11262122531/artifacts/577186863). There should be "release" builds done as part of the next release. `cargo build` is a bit slow on the RPI because RocksDB is slow to build but you can also do cross-compilation, here is [the CI/CD script for that](https://github.com/oxigraph/oxigraph/blob/cbc24950e3d8820603d8d382253bf3b856d56d82/.github/workflows/release.yml#L243).

**Comment by cravishe** [comment]
That's great. Thanks for the information
